src/edu/hillel/Lesson9/Variant_1/Box.java

```diff
@@ -0,0 +1,94 @@
+package edu.hillel.Lesson9.Variant_1;
+
+import edu.hillel.Lesson9.Fruit;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class Box <T extends Fruit>{
+    private final ArrayList<T> fruits;
+    final private T fruitType;
+
+    public Box(T fruitType) {
+        fruits=new ArrayList<>();
+        this.fruitType=fruitType;
+    }
+
+    public ArrayList<T> getFruits() {
+        return fruits;
+    }
+
+    public void fruitAdd (T fruitToAdd) {
+        fruits.add(fruitToAdd);
+    }
+
+    public void fruitsAdd(ArrayList<T> fruitsToAdd){
+        fruits.addAll(fruitsToAdd);
+    }
+    public void fruitsAddIter(ArrayList<T> fruitsToAdd){
+        //second type of method using iteration and method of add by one
+        for (T fruit : fruitsToAdd) {
+            fruitAdd(fruit);
+        }
+    }
+    public void fruitsRemoveAll(){
+        fruits.clear();
+    }
+    public double getWeight() {
+        double totalWeight=0;
+        for (T fruit : fruits) {
+            totalWeight += fruit.getWeight();
+        }
+        return totalWeight;
+    }
+
+    public int getCount(){
+        return fruits.size();
+    }
+
+    public boolean compare(Box<?> box){
+        return (getWeight()==box.getWeight());
+    }
+
+    public void merge(Box<? extends Fruit> box){
+        if (box.fruits.isEmpty()) {
+            System.out.println("Input box is empty! Nothing to merge");
+        }
+        else if (!compareFruitTypes(box)) {
```

Comment:
What sense of this condition?

Reply:
In first variant I made boxes of specific type. And in merge method 
1) if box is empty (source box) - I print message
2) I made method - compare types
    private boolean compareFruitTypes(Box<? extends Fruit> o) {
        return (getFruitType().getClass()==o.getFruitType().getClass());
    }
and in case of incomparable type - I also print message
3) And only if source box not empty and get fruits with type as in target box - I merge from source to target

But in variant 2 I think realization is more nice and correctly. Variant 1 I added only to see..